html/mycakeapp/src/Controller/ReservesController.php

```diff
@@ -217,4 +228,163 @@ public function checkdetail()
+            if ($usageOfPoints === 'no_use_point') {
+                $_SESSION['payment']['creditcard_id'] = $useCreditcard;
+                $_SESSION['payment']['use_point'] = 0;
+                return $this->redirect(['action' => 'checkpayment']);
+            }
+            $usePoint = $this->request->data['use_point'];
+            if ($usePoint > $totalOfOwnPoints) {
+                $TooManyPointsError = '利用できるポイント以下の数字を入力してください';
+                $this->set(compact('TooManyPointsError'));
+            }
+            if ($reserve->execute($this->request->data) && empty($TooManyPointsError)) { //バリデーションOKなら保存
+                $_SESSION['payment']['creditcard_id'] = $useCreditcard;
+                $_SESSION['payment']['use_point'] = $usePoint;
+                return $this->redirect(['action' => 'checkpayment']);
+            }
+        }
+        $this->set(compact('title', 'reserve', 'cardsInfoOwn', 'totalOfOwnPoints'));
+    }
+    public function checkpayment()
+    {
+        $this->viewBuilder()->setLayout('frame-title');
+        $title = '決済概要';
+        if (empty($_SESSION['payment'])) {
+            $this->request->session()->delete('checkdetail');
+            $this->request->session()->delete('payment');
+            return $this->redirect(['controller' => 'error']);
+        }
+        $fee = $_SESSION['fee'];
+        $discountName = $_SESSION['discount_name'];
+
+        $securityKey = Configure::read('key');
+        $securitySalt = Configure::read('salt');
+        $useCreditcard = Security::decrypt($_SESSION['payment']['creditcard_id'], $securityKey, $securitySalt);
+
+        $displayedAmount = $_SESSION['displayed_amount'];
+        $usePoint = $_SESSION['payment']['use_point'];
+        $subTotal = $fee - $displayedAmount - $usePoint;
+        $taxRate = $this->Taxes->find()->select(['tax_rate'])->where(['id' => 1])->first()->tax_rate;
+        $tax = floor($subTotal * ($taxRate / 100));
+        $purchasePrice = $subTotal + $tax;
+
+        $this->set(compact('fee', 'discountName', 'displayedAmount', 'subTotal', 'tax', 'purchasePrice'));
+
+        $paymentsEntity = $this->Payments->newEntity();
+        if ($this->request->is('post')) {
+            $today = Time::now();
+            $memberId = $this->Auth->user('id');
+
+            $payments = [
+                'member_id' => (int)$memberId,
+                'schedule_id' => (int)$_SESSION['schedule_id'],
+                'column_number' => $_SESSION['column_number'],
+                'record_number' => $_SESSION['record_number'],
+                'creditcard_id' => $useCreditcard,
+                'purchase_price' => (int)$purchasePrice,
+                'is_cancelled' => 0,
+                'created_at' => $today,
+                'updated_at' => $today
+            ];
+            $pointRate = $this->PointRates->find()->select(['point_rate'])->where(['id' => 1])->first()->point_rate;
+            $givePoint = ($fee * ($pointRate / 100));
```

Comment:
付与ポイントは決済額の10％なので(基本設計書P18参照)
```suggestion
            $givePoint = floor(($purchasePrice * ($pointRate / 100)));
```
が正しい形だと思います！

Reply:
こちらで修正いたしました 793f381